seagap/configs/solver.py

```diff
@@ -0,0 +1,142 @@
+"""solver.py
+
+The solver module containing base models for
+solver configuration
+"""
+from typing import Any, List, Literal, Optional
+from uuid import uuid4
+
+from pydantic import BaseModel, Field, PrivateAttr, validator
+
+
+class ReferenceEllipsoid(BaseModel):
+    """Reference ellipsoid base model"""
+
+    semi_major_axis: float = Field(..., description="Semi-major axis (m)")
+    reverse_flattening: float = Field(..., description="Reverse flattening")
+    eccentricity: Optional[float] = Field(
```

Comment:
Since this is computed, is there any way to disallow setting it manually to avoid confusion>

Reply:
Great question... unfortunately that's still WIP upstream. See https://github.com/pydantic/pydantic/pull/2625